### DISCUSSION TITLE
Setup with a second X-Touch

### DISCUSSION THREAD

**Discussion by lamusician** [bjoluc/cubase-mcu-midiremote#2] [Q&A]
I must have missed something here unless its the fact that i called the 2nd Xtouch an "extender"? 
Everything works fine but absolutely none of the features is recognized its basically back to where it was with upgraded firmware to the Xtouch.
my sends still show up extended instead of having level and on/off together 
Scribble strips don't change or the added benefit of text 

Basically every reason i added the script in the first place without the benefits

Any ideas?
keeping in mind i do know how to follow directions 
I tried the unaltered edited file and the original that only had the "main" unit

Seems like it would be great

**Comment by bjoluc** [comment]
Hi @lamusician,

> Everything works fine but absolutely none of the features is recognized its basically back to where it was with upgraded firmware to the Xtouch.
> my sends still show up extended instead of having level and on/off together

could you please check whether any old Mackie Control devices are configured in the studio setup?

> I must have missed something here unless its the fact that i called the 2nd Xtouch an "extender"?

I suspect an X-Touch configured as an X-Touch Extender won't process SysEx messages as extenders work with slightly different SysEx headers. Please try `decives: ["main", "main"],`. The button section of the left X-Touch will be unused in this setting anyway.

> keeping in mind i do know how to follow directions

That sounds like a good prerequisite :sweat_smile:

> I tried the unaltered edited file and the original that only had the "main" unit

Did the original file work as expected?

**Reply by lamusician** [reply]
Thank you so much for your reply. So I did try Try "main" under devices instead of extender which didn't seem to make a difference,  but it didn't break anything either,  so thats where i left it.
I will try this again this morning 

It was file the file I was working with didn't add any benifit and I just had to reconfigure. Maybe I grabbed the wrong file so thanks for the direct link. I will try with only 1 X-touch just to see it come up.

I saw that other script with color commands added but didn't see it in the one I downloaded. Not that I can actually do anything from a code perspective anyway.

I don't remember seeing a .js extension on the file I originally downloaded but the properties showed Java.

Hopefully thats it. I've gotten pretty fast at it now. I will check back with you either way and thanks again for responding

**Reply by lamusician** [reply]
tried again it did not automatically find the MC device upon restarting. 
When i went to add a MC device  manually  it shows up with all my shortcuts already there. 
That shouldn't happen right? 
This was done with only 1 unit and the original file you sent.
Maybe delete cubase preferences?
Attatched 2 screen shots
![Add device 1](https://user-images.githubusercontent.com/126140006/220955763-4c74e8e7-148b-417b-b239-19196feda742.JPG)
![Add device 2](https://user-images.githubusercontent.com/126140006/220955774-ef861aec-84ae-4129-8d35-6af389825b72.JPG)

**Reply by bjoluc** [reply]
Would you mind following the official [setup steps](https://github.com/bjoluc/cubase-xtouch-midiremote#setup) and letting me know if anything is unclear?

**Reply by lamusician** [reply]
I did all of that exactly and I thought your instructions were easy enough.
I deleted all Cubase Default preferences  (renamed to default.old)  and started again still no luck, however.
It did at least start as fresh machines with not of my shortcuts preloaded so that verifies any left over preferences.
I tried using only 1 machine in case that was it.
I currently have the modified file  "main/main" loaded in the folder as stated it doesn't seem to do anything good or bad
  devices: ["main", "main"], This was the only edit i made 

I also have a MIDI remote device folder called "X touch left" with a .json file to map some of the top keys. 
That isn't the issue because i literally pulled  out of the folder when i started from scratch.

Doesn't seem i am going to get this to work which is a real bummer
Should I pull your file out of that folder and let Cubase write another one?
it isn't hurting anything

**Reply by bjoluc** [reply]
I'm sorry for suspecting you didn't follow all steps then – I was just confused by "When i went to add a MC device manually": The script is supposed to be set up in the lower zone MIDI Remote pane, not as a remote device in the studio setup options. I usually don't do this, but: If you're willing to set up a quick face-2-face session to sort things out and save us both some time, feel free to email me. We could share our findings in this thread afterwards.

**Reply by lamusician** [reply]
thank you

**Reply by bjoluc** [reply]
To anyone stumbling upon this thread, the issues were:
* The name of the `xtouch` folder that contains the script must be exactly `xtouch`, `X-Touch` doesn't work – I'll add a note about this to the Readme.
* Cubase on Windows has trouble dealing with two identical USB MIDI devices (they were both named "X-Touch", making only one input show up in the port selection dialogs). The solution was to set one X-Touch up via RTP MIDI so the device name could be freely chosen.
